src/ChaiVM/interpreter/executor.cpp

```diff
@@ -388,12 +392,58 @@ void Executor::string_len(Instruction ins) {
 }
 void Executor::string_slice(Instruction ins) {
     const std::string &str = codeManager_->getStringByStringPoolPos(acc());
+    std::cout << "string_slice" << std::endl;
     acc() = codeManager_->getCnst(codeManager_->addCnstString(
         str.substr((*currentFrame_)[ins.r1],
                    (*currentFrame_)[ins.r2] - (*currentFrame_)[ins.r1])));
     advancePc();
     DO_NEXT_INS();
 }
+void Executor::alloc_ref(Instruction ins) {
+    const Klass &klass = codeManager_->getKlass(ins.immidiate);
+    assert(klass.instanceSize() > 0);
+    memory::LinearAllocator<uint8_t> allocator{objectsBuffer_};
+    auto *object = new (allocator.allocate(klass.instanceSize()))
+        uint8_t[klass.instanceSize()]();
+    auto *pheader = reinterpret_cast<ObjectHeader *>(object);
+    auto *fields = reinterpret_cast<chsize_t *>(object + sizeof(*pheader));
+    pheader->size_ = klass.instanceSize();
+    pheader->klassId_ = ins.immidiate;
+    for (int i = 0; i < klass.nFields(); ++i) {
+        assert(fields[i] == 0);
+    }
+    acc() = std::bit_cast<chsize_t>(object);
+    advancePc();
+    DO_NEXT_INS()
+}
+void Executor::mov_ref(Instruction ins) {
+    std::cout << ins.operation << ": mov_ref is not implemented" << std::endl;
+    assert(1 == 0);
+}
+void Executor::ldra_ref(Instruction ins) {
+    acc() = (*currentFrame_)[ins.r1];
+    advancePc();
+    DO_NEXT_INS()
+}
+void Executor::star_ref(Instruction ins) {
+    (*currentFrame_)[ins.r1] = acc();
+    advancePc();
+    DO_NEXT_INS()
+}
```

Comment:
@levBagryansky what is the idea of this methods? how this methods differ from regular `mov`, `ldra`, `star`? 

Do you need it only to distinguish ref operation from non ref to count references?